proof/crefine/AARCH64/Fastpath_Equiv.thy

```diff
@@ -1067,11 +1088,11 @@ crunch only_cnode_caps[wp]: doFaultTransfer "\<lambda>s. P (only_cnode_caps (cte
 lemma tcbSchedDequeue_rewrite_not_queued:
   "monadic_rewrite True False (tcb_at' t and obj_at' (Not \<circ> tcbQueued) t)
      (tcbSchedDequeue t) (return ())"
-  apply (simp add: tcbSchedDequeue_def when_def)
-  apply (monadic_rewrite_l monadic_rewrite_if_l_False \<open>wp threadGet_const\<close>)
-   apply (monadic_rewrite_symb_exec_l, rule monadic_rewrite_refl)
-   apply wp+
-  apply (clarsimp simp: o_def obj_at'_def)
+  apply (simp add: tcbSchedDequeue_def)
+  apply monadic_rewrite_symb_exec_l
+    apply (monadic_rewrite_symb_exec_l_known False, simp)
+     apply (rule monadic_rewrite_refl)
+    apply (wpsimp wp: threadGet_const)+
```

Comment:
Something about this strikes me as automation failing or being disabled compared to what we had previously. Did something in the spec get swapped around? We're no longer doing a generic step with a provided rule for threadGet, and it seems we need to walk to it manually now.

Reply:
The spec for `tcbSchedDequeue` has changed. It now uses the prev and next pointers, but maybe it is the addition of the `stateAssert` that is causing the problem here. I think you mentioned something before about functions which do not change the state and whose results are not used later in the monadic definition tripping up the automation somehow